bundlewatch stopped working

it seems that the bundle watch stopped working, it's not published anymore so even where there are no changes in size it shows increase `(+11.87KB, -0B, +16.3%)`

Not sure when I started to fail.

This is the Azure DevOps logs:

```
nps is executing `build.size` : (cross-env BUNDLEWATCH_GITHUB_TOKEN='undefined' CI_REPO_OWNER='isomorphic-git' CI_REPO_NAME='isomorphic-git' CI_COMMIT_SHA='undefined' CI_BRANCH='undefined' CI_BRANCH_BASE='undefined' bundlewatch) || echo "Optional command 'cross-env BUNDLEWATCH_GITHUB_TOKEN=\'undefined\' CI_REPO_OWNER=\'isomorphic-git\' CI_REPO_NAME=\'isomorphic-git\' CI_COMMIT_SHA=\'undefined\' CI_BRANCH=\'undefined\' CI_BRANCH_BASE=\'undefined\' bundlewatch' failed".
[ERROR] GitHubService HTTP_401 :: Bad credentials
[INFO] Retrieving comparison
[ERROR] Unable to fetch fileDetails for baseBranch=undefined from https://service.bundlewatch.io/store code=ERR_BAD_REQUEST
[INFO] undefined is not a branch to track, no results saved
[ERROR] GitHubService HTTP_401 :: Bad credentials
```

--- a/package-scripts.cjs
+++ b/package-scripts.cjs
@@ -25,37 +25,43 @@ const timeout15 = (command) =>
  * @returns {string}
  */
 const bundlewatchEnvironmentVariables = () => {
-  const options = [
-    `BUNDLEWATCH_GITHUB_TOKEN='${process.env.BUNDLEWATCH_GITHUB_TOKEN}'`,
-    `CI_REPO_OWNER='isomorphic-git'`,
-    `CI_REPO_NAME='isomorphic-git'`,
-  ]
+  // Interpolating an unset variable yields the literal string 'undefined',
+  // which bundlewatch then treats as a real value. Skip the option instead.
+  const options = []
+  const set = (name, value) => {
+    if (value !== undefined && value !== '') options.push(`${name}='${value}'`)
+  }
+
+  set('BUNDLEWATCH_GITHUB_TOKEN', process.env.BUNDLEWATCH_GITHUB_TOKEN)
+  set('CI_REPO_OWNER', 'isomorphic-git')
+  set('CI_REPO_NAME', 'isomorphic-git')
 
   // Azure DevOps Pipeline is not detected by bundlewatch (which uses ci-env).
   if (process.env.SYSTEM_COLLECTIONURI !== undefined) {
-    options.push(
-      `CI_COMMIT_SHA='${process.env.TRAVIS_PULL_REQUEST_SHA}'`,
-      `CI_BRANCH='${process.env.SYSTEM_PULLREQUEST_SOURCEBRANCH}'`,
-      `CI_BRANCH_BASE='${process.env.SYSTEM_PULLREQUEST_TARGETBRANCH}'`
+    // On a pull request build Azure checks out a merge commit, so report the
+    // head of the source branch, which is the commit the status belongs to.
+    set(
+      'CI_COMMIT_SHA',
+      process.env.SYSTEM_PULLREQUEST_SOURCECOMMITID ??
+        process.env.BUILD_SOURCEVERSION
     )
+    set('CI_BRANCH', process.env.SYSTEM_PULLREQUEST_SOURCEBRANCH)
+    set('CI_BRANCH_BASE', process.env.SYSTEM_PULLREQUEST_TARGETBRANCH)
   }
 
   // GitHub is not detected well using bundlewatch@0.2.5.
   else if (process.env.GITHUB_SHA) {
-    options.push(`CI_COMMIT_SHA='${process.env.GITHUB_SHA}'`)
+    set('CI_COMMIT_SHA', process.env.GITHUB_SHA)
 
     // Get the current and base branch for pull requests
     if (process.env.GITHUB_EVENT_NAME === 'pull_request') {
-      options.push(
-        `CI_BRANCH='${process.env.GITHUB_HEAD_REF}'`,
-        `CI_BRANCH_BASE='${process.env.GITHUB_BASE_REF}'`
-      )
+      set('CI_BRANCH', process.env.GITHUB_HEAD_REF)
+      set('CI_BRANCH_BASE', process.env.GITHUB_BASE_REF)
     }
     // Otherwise, get the current branch. This handles push and workflow_dispatch events.
     // This does not handle tags (refs/tags/<tag_name>).
-    else if (process.env.GITHUB_REF.startsWith('refs/heads/')) {
-      const branch = process.env.GITHUB_REF.replace('refs/heads/')
-      options.push(`CI_BRANCH='${branch}'`)
+    else if (process.env.GITHUB_REF?.startsWith('refs/heads/')) {
+      set('CI_BRANCH', process.env.GITHUB_REF.replace('refs/heads/', ''))
     }
   }
 

